oceannavigator/frontend/src/components/Class4Window.jsx

```diff
@@ -1,197 +1,159 @@
-import React from "react";
+import React, { useState, useEffect, useRef } from "react";
 import { Accordion, Card, Row, Col } from "react-bootstrap";
-import PlotImage from "./PlotImage.jsx";
 import ComboBox from "./ComboBox.jsx";
 import CheckBox from "./lib/CheckBox.jsx";
 import ImageSize from "./ImageSize.jsx";
+import PlotImage from "./PlotImage.jsx";
 import PropTypes from "prop-types";
-
-import { withTranslation } from "react-i18next";
-
-class Class4Window extends React.Component {
-  constructor(props) {
-    super(props);
-
-    // Track if mounted to prevent no-op errors with the Ajax callbacks.
-    this._mounted = false;
-
-    this.state = {
-      forecast: "best",
-      showmap: false,
-      climatology: false,
-      error: "none",
-      size: "10x7",
-      dpi: 144,
-      models: [],
+import { useTranslation } from "react-i18next";
+
+const Class4Window = ({ dataset, class4type, plotData, init = {}, action }) => {
+  const { t: _ } = useTranslation();
+  // Track if mounted to prevent no-op errors with the Ajax callbacks.
+  const isMountedRef = useRef(false);
+
```

Comment:
I think this was necessary for class component lifecycles but can probably be removed for functional ones.